core/.js/src/main/scala/io/udash/utils/FileService.scala

```diff
@@ -0,0 +1,45 @@
+package io.udash.utils
+
+import java.io.IOException
+
+import org.scalajs.dom._
+import scalatags.JsDom.all._
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.typedarray._
+
+import scala.concurrent._
+import scala.util.Try
+
+object FileService {
+  def save(filename: String, bytes: Array[Byte], mimeType: String = "application/octet-stream"): Unit = {
+    val jsBytes = Array[js.Any](bytes.toTypedArray).toJSArray
+    val blob = new Blob(jsBytes, BlobPropertyBag(mimeType))
+    val url = URL.createObjectURL(blob)
+    val download = attr("download")
+
+    val anchor = a(href := url, download := filename, style := "display: none")(filename).render
+    document.body.appendChild(anchor)
+    anchor.click()
+    document.body.removeChild(anchor)
+    URL.revokeObjectURL(url)
+  }
+
+  def load(file: File): Future[Array[Byte]] = {
+    val fileReader = new FileReader()
+    val promise = Promise[Array[Byte]]()
+
+    fileReader.onerror = (e: Event) =>
+      promise.failure(new IOException(e.toString))
+
+    fileReader.onload = (_: UIEvent) =>
+      promise.complete(Try(
+        new Int8Array(fileReader.result.asInstanceOf[ArrayBuffer]).toArray
+      ))
+
+    fileReader.readAsArrayBuffer(file)
+
+    promise.future
+  }
+}
```

Comment:
This file needs some docs - these are not very straightforward APIs and the use case can be hard to deduce, leading to noone finding / using this.